rpmsg framing, stats reporting

silverjam opened the pull request.

silverjam committed:
rpmsg_piksi: log error if we drop rx data

silverjam committed:
rpmsg: improve start-up script formatting

silverjam committed:
rpmsg_stats: initial stats daemon

anonymous-travis-swiftnav commented:
## HITL smoke tests: v2.1.0-develop-2019031522-6-g1879cdf3

These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.

### job status
+ [live-roof-650-townsend runs](https://gnss-analysis.swiftnav.com/jobs/id=951623)
+ [live-roof-650-townsend-dropouts-zero-baseline runs](https://gnss-analysis.swiftnav.com/jobs/id=951624)

### gnss-analysis results
At least one run must complete for these links to have data.

#### passfail
+ [live-roof-650-townsend](https://gnss-analysis.swiftnav.com/metrics_preset=pass_fail&scenario=live-roof-650-townsend&summary_type=q50&groupby_key=firmware&firmware_versions=v2.1.0-develop-2019031522-6-g1879cdf3&build_type=buildroot_pr&display_type=table)
+ [live-roof-650-townsend-dropouts-zero-baseline](https://gnss-analysis.swiftnav.com/metrics_preset=pass_fail&scenario=live-roof-650-townsend-dropouts-zero-baseline&summary_type=q50&groupby_key=firmware&firmware_versions=v2.1.0-develop-2019031522-6-g1879cdf3&build_type=buildroot_pr&display_type=table)

#### detailed
+ [live-roof-650-townsend](https://gnss-analysis.swiftnav.com/metrics_preset=detailed&scenario=live-roof-650-townsend&summary_type=q50&groupby_key=firmware&firmware_versions=v2.1.0-develop-2019031522-6-g1879cdf3&build_type=buildroot_pr&display_type=table)
+ [live-roof-650-townsend-dropouts-zero-baseline](https://gnss-analysis.swiftnav.com/metrics_preset=detailed&scenario=live-roof-650-townsend-dropouts-zero-baseline&summary_type=q50&groupby_key=firmware&firmware_versions=v2.1.0-develop-2019031522-6-g1879cdf3&build_type=buildroot_pr&display_type=table)

anonymous-travis-swiftnav commented:
## v2.1.0-develop-2019031522-6-g1879cdf3
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v2.1.0-develop-2019031522-6-g1879cdf3)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v2.1.0-develop-2019031522-6-g1879cdf3/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v2.1.0-develop-2019031522-6-g1879cdf3

silverjam committed:
rpmsg_stats: use nmea rpmsg port for stats

silverjam committed:
nmea: pass-thru NMEA GSV message from firmware

temporary hack that can be dropped one we have an SBP message that
includes the relavent info to generate the GSV message

silverjam committed:
fetch_firmware: development pfwp

anonymous-travis-swiftnav commented:
## HITL smoke tests: v2.1.0-develop-2019031522-12-ga0ea8b16

These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.

### job status
+ [live-roof-650-townsend runs](https://gnss-analysis.swiftnav.com/jobs/id=956037)
+ [live-roof-650-townsend-dropouts-zero-baseline runs](https://gnss-analysis.swiftnav.com/jobs/id=956038)

### gnss-analysis results
At least one run must complete for these links to have data.

#### passfail
+ [live-roof-650-townsend](https://gnss-analysis.swiftnav.com/metrics_preset=pass_fail&scenario=live-roof-650-townsend&summary_type=q50&groupby_key=firmware&firmware_versions=v2.1.0-develop-2019031522-12-ga0ea8b16&build_type=buildroot_pr&display_type=table)
+ [live-roof-650-townsend-dropouts-zero-baseline](https://gnss-analysis.swiftnav.com/metrics_preset=pass_fail&scenario=live-roof-650-townsend-dropouts-zero-baseline&summary_type=q50&groupby_key=firmware&firmware_versions=v2.1.0-develop-2019031522-12-ga0ea8b16&build_type=buildroot_pr&display_type=table)

#### detailed
+ [live-roof-650-townsend](https://gnss-analysis.swiftnav.com/metrics_preset=detailed&scenario=live-roof-650-townsend&summary_type=q50&groupby_key=firmware&firmware_versions=v2.1.0-develop-2019031522-12-ga0ea8b16&build_type=buildroot_pr&display_type=table)
+ [live-roof-650-townsend-dropouts-zero-baseline](https://gnss-analysis.swiftnav.com/metrics_preset=detailed&scenario=live-roof-650-townsend-dropouts-zero-baseline&summary_type=q50&groupby_key=firmware&firmware_versions=v2.1.0-develop-2019031522-12-ga0ea8b16&build_type=buildroot_pr&display_type=table)

anonymous-travis-swiftnav commented:
## v2.1.0-develop-2019031522-12-ga0ea8b16
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v2.1.0-develop-2019031522-12-ga0ea8b16)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v2.1.0-develop-2019031522-12-ga0ea8b16/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v2.1.0-develop-2019031522-12-ga0ea8b16

silverjam committed:
endpoint_router: cache reject rules

Reject rules need to be captured for chains that are REJECT -> REJECT ->
* -> ACCEPT so that we can correctly ignore certain incoming data.

For this case we cache the match, but don't specify a port.

anonymous-travis-swiftnav commented:
## HITL smoke tests: v2.1.0-develop-2019031522-13-gcb30aa1a

These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.

### job status
+ [live-roof-650-townsend runs](https://gnss-analysis.swiftnav.com/jobs/id=956846)
+ [live-roof-650-townsend-dropouts-zero-baseline runs](https://gnss-analysis.swiftnav.com/jobs/id=956847)

### gnss-analysis results
At least one run must complete for these links to have data.

#### passfail
+ [live-roof-650-townsend](https://gnss-analysis.swiftnav.com/metrics_preset=pass_fail&scenario=live-roof-650-townsend&summary_type=q50&groupby_key=firmware&firmware_versions=v2.1.0-develop-2019031522-13-gcb30aa1a&build_type=buildroot_pr&display_type=table)
+ [live-roof-650-townsend-dropouts-zero-baseline](https://gnss-analysis.swiftnav.com/metrics_preset=pass_fail&scenario=live-roof-650-townsend-dropouts-zero-baseline&summary_type=q50&groupby_key=firmware&firmware_versions=v2.1.0-develop-2019031522-13-gcb30aa1a&build_type=buildroot_pr&display_type=table)

#### detailed
+ [live-roof-650-townsend](https://gnss-analysis.swiftnav.com/metrics_preset=detailed&scenario=live-roof-650-townsend&summary_type=q50&groupby_key=firmware&firmware_versions=v2.1.0-develop-2019031522-13-gcb30aa1a&build_type=buildroot_pr&display_type=table)
+ [live-roof-650-townsend-dropouts-zero-baseline](https://gnss-analysis.swiftnav.com/metrics_preset=detailed&scenario=live-roof-650-townsend-dropouts-zero-baseline&summary_type=q50&groupby_key=firmware&firmware_versions=v2.1.0-develop-2019031522-13-gcb30aa1a&build_type=buildroot_pr&display_type=table)

anonymous-travis-swiftnav commented:
## v2.1.0-develop-2019031522-13-gcb30aa1a
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v2.1.0-develop-2019031522-13-gcb30aa1a)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v2.1.0-develop-2019031522-13-gcb30aa1a/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v2.1.0-develop-2019031522-13-gcb30aa1a

silverjam committed:
endpoint_router: drop framer support

Having framer support is trickier than I initial thought (exact
reasons are unclear) but this was occasionally causing us to drop
packets under heavy load.

As such, just remove it for now and put framing back in the
endpoint_adapter.

silverjam committed:
endpoint_adapter: don't drop data

We drop data if the remote client gets behind, it turns out that the
console can demonstrate how this can cause extremely bad behavior
in a client... when the console gets behind we'll start dropping
a few bytes from each send that arrives to endpoint_adapter (10-30
bytes).

Instead of doing this we'll just do our best in this situation and
report that the blocking is happening. In the case were we get really
behind (160k) the endpoint_router will eventually close the socket and
cause us to disconnect from the remote client.

silverjam committed:
endpoint_adapter: add "bypass" pub/sub support

silverjam committed:
sbp_rx: hack to re-use existing state with a different endpoint

Next step is to allow sbp_rx objects to be associated with multiple
endpoints.

silverjam committed:
sbp_tx: support for dumping output to file

can probably be dumped later

silverjam committed:
libpiksi: add ability debug recv_impl alone

silverjam committed:
ports_daemon: add bypass switches to endpoints

silverjam committed:
sbp_fileio: add "bypass" pub/sub ports

silverjam committed:
sbp_rtcm3_bridge: log if sbp_tx_send_from fails

silverjam committed:
drop: dump sbp_rtcm3_bridge output

anonymous-travis-swiftnav commented:
## HITL smoke tests: v2.1.0-develop-2019031522-27-gca391736

These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.

### job status
+ [live-roof-650-townsend runs](https://gnss-analysis.swiftnav.com/jobs/id=961434)
+ [live-roof-650-townsend-dropouts-zero-baseline runs](https://gnss-analysis.swiftnav.com/jobs/id=961435)

### gnss-analysis results
At least one run must complete for these links to have data.

#### passfail
+ [live-roof-650-townsend](https://gnss-analysis.swiftnav.com/metrics_preset=pass_fail&scenario=live-roof-650-townsend&summary_type=q50&groupby_key=firmware&firmware_versions=v2.1.0-develop-2019031522-27-gca391736&build_type=buildroot_pr&display_type=table)
+ [live-roof-650-townsend-dropouts-zero-baseline](https://gnss-analysis.swiftnav.com/metrics_preset=pass_fail&scenario=live-roof-650-townsend-dropouts-zero-baseline&summary_type=q50&groupby_key=firmware&firmware_versions=v2.1.0-develop-2019031522-27-gca391736&build_type=buildroot_pr&display_type=table)

#### detailed
+ [live-roof-650-townsend](https://gnss-analysis.swiftnav.com/metrics_preset=detailed&scenario=live-roof-650-townsend&summary_type=q50&groupby_key=firmware&firmware_versions=v2.1.0-develop-2019031522-27-gca391736&build_type=buildroot_pr&display_type=table)
+ [live-roof-650-townsend-dropouts-zero-baseline](https://gnss-analysis.swiftnav.com/metrics_preset=detailed&scenario=live-roof-650-townsend-dropouts-zero-baseline&summary_type=q50&groupby_key=firmware&firmware_versions=v2.1.0-develop-2019031522-27-gca391736&build_type=buildroot_pr&display_type=table)

anonymous-travis-swiftnav commented:
## v2.1.0-develop-2019031522-27-gca391736
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v2.1.0-develop-2019031522-27-gca391736)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v2.1.0-develop-2019031522-27-gca391736/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v2.1.0-develop-2019031522-27-gca391736

silverjam committed:
sketch of making sbp_rx hold multiple sockets

silverjam committed:
sbp_rx: revert support for multiple endpoints

silverjam committed:
Enable sbp_pubsub to launch server sockets

silverjam committed:
sbp_fileio_daemon: operate on lists of sockets

Allow sbp_fileio_daemon to operate on lists of sockets (and for some of
those sockets to be server sockets).

silverjam committed:
ports_daemon: hook-up to fileio ports

silverjam committed:
endpoint_adapter: change terminology from bypass to alt (needs reorder with prev commit)

anonymous-travis-swiftnav commented:
## HITL smoke tests: v2.1.0-develop-2019032221-28-gf5b8b2e9
There was an error using the HITL API to kick off smoke tests for this commit.
See https://travis-ci.org/swift-nav/piksi_buildroot/builds/511782602

anonymous-travis-swiftnav commented:
## v2.1.0-develop-2019032221-28-gf5b8b2e9
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v2.1.0-develop-2019032221-28-gf5b8b2e9)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v2.1.0-develop-2019032221-28-gf5b8b2e9/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v2.1.0-develop-2019032221-28-gf5b8b2e9
